include/oneapi/mkl/sparse_blas/detail/sparse_blas_ct.hxx

```diff
@@ -0,0 +1,121 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+// This file is meant to be included in each backend sparse_blas_ct.hpp files
+// Each function calls the implementation from onemkl_sparse_blas_backends.hxx
+
+#ifndef BACKEND
+#error "BACKEND is not defined"
+#endif
+
+inline void init_matrix_handle(backend_selector<backend::BACKEND> selector,
+                               matrix_handle_t *p_handle) {
+    BACKEND::init_matrix_handle(selector.get_queue(), p_handle);
+}
+
+inline sycl::event release_matrix_handle(backend_selector<backend::BACKEND> selector,
```

Comment:
How heavy is `backend_selector<backend::BACKEND>`? This would be invoking the copy constructor of that. Should we maybe pass the selector by reference, `&selector`? The `queue` is normally passed by reference, after all.

Reply:
The [`backend_selector`](https://github.com/oneapi-src/oneMKL/blob/develop/include/oneapi/mkl/detail/backend_selector.hpp) contains a SYCL queue. It is always given as a copy in all other domains.
Whether the queue should be given as a reference or copy is debatable. The SYCL spec guarantees that a queue - and most other SYCL objects - can be copied and "must behave as-if it were the original instance and as-if any action performed on it were also performed on the original instance" (i.e. use a pimpl design).
Regarding performance the compiler is allowed to elide copies so this should not be an issue here.

Reply:
Okay, if this is oneMKL-wide, it makes sense to keep it as is.